src/dom/components/datasources/datetime/datetimesourcefactory.hpp

```diff
@@ -0,0 +1,56 @@
+#ifndef DOT_MESSAGE_DATETIME_SOURCEFACTORY_HPP
+#define DOT_MESSAGE_DATETIME_SOURCEFACTORY_HPP
+
+#include <memory>
+#include <string>
+#include <vector>
+#include <regex>
+
+#include "dom/components/datasources/datetimesource.hpp"
+
+namespace macsa
+{
+	namespace dot
+	{
+		namespace datetime
+		{
+			/**
+			 * @brief Factory class to parse format from string to vector<DateTimeSource*>
+			 */
+			class DateTimeSourceFactory
+			{
+				public:
+					explicit DateTimeSourceFactory() = default;
+					virtual ~DateTimeSourceFactory() = default;
+
+					/**
+					 * @brief Convert format specifiers into the corresponding DateTimeSource sub-class
+					 */
+					std::vector<std::unique_ptr<DateTimeSource>> parseFormatRegex(const std::string& format);
```

Comment:
```suggestion
					std::vector<std::unique_ptr<DateTimeSource>> ParseFormatRegex(const std::string& format);
```

Caldria afegir un unit test per verificar que funciona segons esperem

Reply:
fet